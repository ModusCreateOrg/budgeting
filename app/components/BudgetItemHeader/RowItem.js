// @flow
import * as React from 'react';
import styles from './style.scss';

type RowItemProps = {
  label: string,
  children: React.Element<any>,
};

const RowItem = ({ label, children }: RowItemProps) => (
  <div className={styles.rowItem}>
    <div className={styles.label}>{label} </div>
    <div className={styles.content}>{children} </div>
  </div>
);

export default RowItem;
