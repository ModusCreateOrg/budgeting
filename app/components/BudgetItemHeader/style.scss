    @import 'theme/variables';
    .budgetItemDetails {
        background-color: $verylightgray;
        .rowItem {
            display: flex;
            flex-wrap: nowrap;
            border: $border;
            .label {
                padding: 10px;
                background-color: $gray;
                min-width: 140px;
            }
            .content {
                align-self: center;
                width: 100%;
                margin-left: 10px;
                word-wrap: break-word;
            }
            %flow {
                font-size: 20px;
                font-weight: 500;

                &:after{
                    content: '%';
                }

            }
            .in-flow {
                @extend %flow;
                color: $green;
                
                &:before{
                    content: '+ ';
                }
            }
            .out-flow {
                @extend %flow;
                color: $red;

                &:before{
                    content: '- ';
                }
            }
        }
    }