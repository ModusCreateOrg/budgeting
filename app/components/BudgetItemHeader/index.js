// @flow
import * as React from 'react';
import { roundUp } from 'utils/arithmetic';
import type { Transaction } from 'modules/transactions';
import RowItem from './RowItem';
import styles from './style.scss';

type BudgetItemHeaderProps = {
  transaction: Transaction,
  balance: number,
};

const BudgetItemHeader = ({ transaction, balance }: BudgetItemHeaderProps) => {
  const contributionPercentage = Math.abs(transaction.value / balance) * 100;
  const roundedPercentage = roundUp(contributionPercentage, 2);
  const flow = `${transaction.value > 0 ? 'in' : 'out'}-flow`;
  return (
    <div className={styles.budgetItemDetails}>
      <RowItem label="Item Description">{transaction.description}</RowItem>
      <RowItem label="Contribution">
        <div className={styles[flow]}>{roundedPercentage}</div>
      </RowItem>
    </div>
  );
};

export default BudgetItemHeader;
