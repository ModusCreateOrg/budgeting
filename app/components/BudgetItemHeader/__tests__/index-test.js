import React from 'react';
import renderer from 'react-test-renderer';
import BudgetItemHeader from 'components/BudgetItemHeader';

it('component renders acceptably for inflow', () => {
  const transaction = {
    id: 1,
    description: 'Paycheck',
    value: 5000,
    categoryId: 15,
  };

  const balance = 6500;

  const tree = renderer.create(<BudgetItemHeader transaction={transaction} balance={balance} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('component renders acceptably for outflow', () => {
  const transaction = {
    id: 4,
    description: 'Paycheck',
    value: -485,
    categoryId: 12,
  };

  const balance = 2300;

  const tree = renderer.create(<BudgetItemHeader transaction={transaction} balance={balance} />).toJSON();
  expect(tree).toMatchSnapshot();
});
