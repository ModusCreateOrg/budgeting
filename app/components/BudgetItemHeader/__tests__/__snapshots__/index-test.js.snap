// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders acceptably for inflow 1`] = `
<div
  className="budgetItemDetails"
>
  <div
    className="rowItem"
  >
    <div
      className="label"
    >
      Item Description
       
    </div>
    <div
      className="content"
    >
      Paycheck
       
    </div>
  </div>
  <div
    className="rowItem"
  >
    <div
      className="label"
    >
      Contribution
       
    </div>
    <div
      className="content"
    >
      <div
        className="in-flow"
      >
        76.92
      </div>
       
    </div>
  </div>
</div>
`;

exports[`component renders acceptably for outflow 1`] = `
<div
  className="budgetItemDetails"
>
  <div
    className="rowItem"
  >
    <div
      className="label"
    >
      Item Description
       
    </div>
    <div
      className="content"
    >
      Paycheck
       
    </div>
  </div>
  <div
    className="rowItem"
  >
    <div
      className="label"
    >
      Contribution
       
    </div>
    <div
      className="content"
    >
      <div
        className="out-flow"
      >
        21.09
      </div>
       
    </div>
  </div>
</div>
`;
